components/dashboard/src/settings/Preferences.tsx

```diff
@@ -30,6 +30,14 @@ export default function Preferences() {
         const settings = additionalData.ideSettings || {};
         settings.defaultIde = value;
         additionalData.ideSettings = settings;
+        getGitpodService().server.trackEvent({
+            event: "ide_configuration_changed",
+            properties: {
+                useDesktopIde,
+                defaultIde: value,
+                defaultDesktopIde
+            },
+        });
```

Comment:
Will this apply the userId somewhere else, or do we need to do it explicitly here? 

CC: @akosyakov

Reply:
Looks like it's coming through as an anonymous event. 

![image](https://user-images.githubusercontent.com/5528307/146357393-3d0bf189-670a-4ba0-813c-0b51dd8d202b.png)

Reply:
> Don't worry, the user part is handled on the server side

Exactly! The user ID is added for all analytical events sent to the server if the user is authenticated at the time of the event.

Reply:
All events enriched with user Id on the server.